tests/Unit/Elliptic/Systems/Xcts/Events/Test_ObserveAdmIntegrals.cpp

```diff
@@ -0,0 +1,163 @@
+      true,
+      {},
+      {},
+      domain::CoordinateMaps::Distribution::Logarithmic};
+  const auto shell_domain = shell.create_domain();
+  const auto& blocks = shell_domain.blocks();
+  const auto& initial_ref_levels = shell.initial_refinement_levels();
+  const auto element_ids = initial_element_ids(initial_ref_levels);
+  const Mesh<3> mesh{polynomial_order + 1, Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto};
+  const Mesh<2> face_mesh{polynomial_order + 1, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+
+  // Initialize "reduced" integral
+  tnsr::I<double, 3> reduced_integral;
+  for (int I = 0; I < 3; I++) {
+    reduced_integral.get(I) = 0.;
+  }
+
+  // Compute integral by summing over each element
+  for (const auto& element_id : element_ids) {
+    // Get element information
+    const auto& current_block = blocks.at(element_id.block_id());
+    const auto current_element = domain::Initialization::create_initial_element(
+        element_id, current_block, initial_ref_levels);
+    const ElementMap<3, Frame::Inertial> logical_to_inertial_map(
+        element_id, current_block.stationary_map().get_clone());
+
+    // Get coordinates
+    const auto logical_coords = logical_coordinates(mesh);
+    const auto inertial_coords = logical_to_inertial_map(logical_coords);
+    const auto inv_jacobian =
+        logical_to_inertial_map.inv_jacobian(logical_coords);
+
+    // Get required fields
+    const auto background_fields = solution.variables(
+        inertial_coords,
+        tmpl::list<
+            Xcts::Tags::ConformalFactor<DataVector>,
+            gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>,
+            gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>,
+            Xcts::Tags::InverseConformalMetric<DataVector, 3, Frame::Inertial>,
+            gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>,
+            gr::Tags::TraceExtrinsicCurvature<DataVector>>{});
+    const auto& conformal_factor =
+        get<Xcts::Tags::ConformalFactor<DataVector>>(background_fields);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& inv_spatial_metric =
+        get<gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& inv_conformal_metric =
+        get<Xcts::Tags::InverseConformalMetric<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& extrinsic_curvature =
+        get<gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>>(
+            background_fields);
+    const auto& trace_extrinsic_curvature =
+        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(background_fields);
+
+    // Compute face normal (related to the conformal metric)
+    auto direction = Direction<3>::upper_zeta();
```

Comment:
I think you're also integrating over internal element boundaries here. Shouldn't the test fail because of that? I may be missing something.

Reply:
It's my understanding that the reduction will call `Events::local_adm_integrals` on every element (even internal elements). Then, within that function, I handle which elements we should skip. Here, my intention was to simulate how the reduction will call that function (on every element). Am I understanding it wrong?

Reply:
Yes that sounds good, thanks for clarifying 👍